Common/BaselineSelection.py

```diff
@@ -207,75 +214,79 @@ def RecoLeptonsSelection(df, apply_filter=True):
+                  && Tau_idDeepTau2017v2p1VSmu >= {WorkingPointsTauVSmu.VLoose.value}
+                  && Tau_idDeepTau2017v2p1VSjet >= {WorkingPointsTauVSjet.VVVLoose.value} )
+             || (    Tau_idDeepTau2018v2p5VSe >= {WorkingPointsTauVSe.VVLoose.value}
+                  && Tau_idDeepTau2018v2p5VSmu >= {WorkingPointsTauVSmu.VLoose.value}
+                  && Tau_idDeepTau2018v2p5VSjet >= {WorkingPointsTauVSjet.VVVLoose.value} )
+           )
+    '''
+    eta_cut = 2.3 if config["deepTauVersion"] == '2p1' else 2.5
+
     df = df.Define("Electron_iso", "Electron_pfRelIso03_all") \
            .Define("Muon_iso", "Muon_pfRelIso04_all") \
            .Define("Tau_iso", f"""-Tau_rawDeepTau{deepTauVersions[config["deepTauVersion"]]}v{config["deepTauVersion"]}VSjet""")
-    #print(f"""-Tau_rawDeepTau{deepTauVersions[config["deepTauVersion"]]}v{config["deepTauVersion"]}VSjet""")
 
-    df = df.Define("Electron_B2_eTau_1", f"Electron_B0 && abs(v_ops::eta(Electron_p4)) < 2.3 && v_ops::pt(Electron_p4) > 10 && Electron_mvaIso_WP80 ")
-    #df = df.Define("Electron_B2_eTau_1", f"Electron_B0 && v_ops::pt(Electron_p4) > 20 && Electron_mvaIso_WP80")
-    #df = df.Define("Electron_B2_eTau_1", f"Electron_B0 && v_ops::pt(Electron_p4) > 20 && Electron_mvaNoIso_WP80 && Electron_pfRelIso03_all<0.1")
-    eta_cut = 2.3 if config["deepTauVersion"] == '2p1' else 2.5
+    ########## channel definition ##########
+    df = df.Define("Electron_B2_eTau_1", f"Electron_B0 && Electron_mvaIso_WP80 ")
+    #  Electron_mvaNoIso_WP80 && Electron_pfRelIso03_all < 0.3
     df = df.Define("Tau_B2_eTau_2", f"""
-        Tau_B0 && v_ops::pt(Tau_p4) > 20 && v_ops::eta(Tau_p4) < {eta_cut}
+        Tau_B0 && v_ops::eta(Tau_p4) < {eta_cut}
         && (Tau_idDeepTau{deepTauVersions[config["deepTauVersion"]]}v{config["deepTauVersion"]}VSe >= {getattr(WorkingPointsTauVSe, config["deepTauWPs"]["eTau"]["VSe"]).value})
         && (Tau_idDeepTau{deepTauVersions[config["deepTauVersion"]]}v{config["deepTauVersion"]}VSmu >= {getattr(WorkingPointsTauVSmu, config["deepTauWPs"]["eTau"]["VSmu"]).value})
     """)
-    '''
-    df = df.Define("Muon_B2_muTau_1", f"""
-        Muon_B0 && v_ops::pt(Muon_p4) > 20 && abs(v_ops::eta(Muon_p4))<2.3 && (
-            ( v_ops::pt(Muon_p4) <= 120  && Muon_tightId && Muon_pfRelIso04_all < 0.15) ||
-            ( v_ops::pt(Muon_p4)>120 && Muon_highPtId && Muon_tkRelIso < 0.15 )
-        )
-    """)
-    '''
+
     df = df.Define("Muon_B2_muTau_1", f"""
-        Muon_B0 && v_ops::pt(Muon_p4) > 15 && abs(v_ops::eta(Muon_p4))<2.3 && (
-            ( Muon_tightId && Muon_pfRelIso04_all < 0.15 )
-        )
+        Muon_B0 &&  ( (Muon_tightId && Muon_pfRelIso04_all < 0.15) || (Muon_highPtId && Muon_tkRelIso < 0.15) )
     """)
     df = df.Define("Tau_B2_muTau_2", f"""
-        Tau_B0 && v_ops::pt(Tau_p4) > 20 && v_ops::eta(Tau_p4) < {eta_cut}
+        Tau_B0 && v_ops::eta(Tau_p4) < {eta_cut}
```

Comment:
- you need to apply 
for the tau leg only VS WPs changing, so you can make it in the loop:
```python
for ch in [ "eTau", "muTau", "tauTau" ]:
  cut_str = f'''Tau_B0
    && (Tau_idDeepTau{deepTauVersions[config["deepTauVersion"]]}v{config["deepTauVersion"]}VSe >= {getattr(WorkingPointsTauVSe, config["deepTauWPs"][ch]["VSe"]).value})
    && (Tau_idDeepTau{deepTauVersions[config["deepTauVersion"]]}v{config["deepTauVersion"]}VSmu >= {getattr(WorkingPointsTauVSmu, config["deepTauWPs"][ch]["VSmu"]).value})'''
  df = df.Define(f'Tau_B2_{ch}_2', cut_str)
  if ch == 'tauTau':
    cut_str += f' && (Tau_idDeepTau{deepTauVersions[config["deepTauVersion"]]}v{config["deepTauVersion"]}VSjet >= {getattr(WorkingPointsTauVSjet, config["deepTauWPs"]["tauTau"]["VSjet"]).value})'
    df = df.Define(f'Tau_B2_{ch}_1', cut_str)
```